### DISCUSSION TITLE
Image layers cache?

### DISCUSSION THREAD

**Discussion by amitkumar2283** [distribution/distribution#4427] [General]
We want to cache layer data to an in-memory/Redis cache. Per [the docs](https://distribution.github.io/distribution/about/configuration/#cache), Distribution only caches metadata. Is there an already explored design pattern around this that we can explore?

Is there a plan to add the capability to cache layer data to Distribution?

**Comment by milosgajdos** [comment]
Yes, the cache only caches metadata. If you are looking for faster distribution of the blobs I'd recommend tapping into some CDN through storage driver middleware. At the moment distribution only provides support for [cloudfront](https://github.com/distribution/distribution/tree/f0bd0f68992377b068f2861eaa0bdfabb3e540a4/registry/storage/driver/middleware/cloudfront), but it shouldn't be too hard to implement your own.
